static/docs/user-guide/remotes/ssh.md

```diff
@@ -0,0 +1,177 @@
+
+```dvc
+$ dvc remote add ssh-remote ssh://user@example.com/path/to/dir
+```
+
+We can modify the remote settings with `dvc remote modify`:
+
+```dvc
+$ dvc remote modify ssh-remote port 2222
+```
+
+The configuration file `.dvc/config` now should have a content like this:
+
+```ini
+['remote "ssh-remote"']
+url = ssh://user@example.com/path/to/dir
+port = 2222
+```
+
+<details>
+
+### Details: All remote options
+
+- `url` - remote location URL.
+
+  ```dvc
+  $ dvc remote modify ssh-remote url ssh://user@example.com:1234/path/to/dir
+  ```
+
+- `user` - username to use to access a remote. The order in which dvc searches
+  for username:
+
+  1. `user` specified in one of the dvc configs;
+  2. `user` specified in the url(e.g. `ssh://user@example.com/path`);
+  3. `user` specified in `~/.ssh/config` for remote host;
+  4. current user;
+
+  ```dvc
+  $ dvc remote modify ssh-remote user myuser
+  ```
+
+- `port` - port to use to access a remote. The order in which dvc searches for
+  port:
+
+  1. `port` specified in one of the dvc configs;
+  2. `port` specified in the url(e.g. `ssh://example.com:1234/path`);
+  3. `port` specified in `~/.ssh/config` for remote host;
+  4. default ssh port 22;
+
+  ```dvc
+  $ dvc remote modify ssh-remote port 2222
+  ```
+
+- `keyfile` - path to private key to use to access a remote.
+
+  ```dvc
+  $ dvc remote modify ssh-remote keyfile /path/to/keyfile
+  ```
+
+- `password` - a private key passphrase or a password to use to use when
+  accessing a remote.
+
+  ```dvc
+  $ dvc remote modify ssh-remote password mypassword
+  ```
+
+- `ask_password` - ask for a private key passphrase or a password to use when
+  accessing a remote.
+
+  ```dvc
+  $ dvc remote modify ssh-remote ask_password true
+  ```
+
+- `gss_auth` - use Generic Security Services authentication if available on host
+  (for example,
+  [with kerberos](https://en.wikipedia.org/wiki/Generic_Security_Services_Application_Program_Interface#Relationship_to_Kerberos)).
+  Using this option requires `paramiko[gssapi]`, which is currently only
+  supported by our pip package, and could be installed with
+  `pip install 'dvc[ssh_gssapi]'`. Other packages (Conda, Windows, and MacOS
+  PKG) do not support it.
+
+  ```dvc
+  $ dvc remote modify ssh-remote gss_auth true
+  ```
+
+</details>
+
+<details>
+
+### Tip: Keep the password private
+
+In general it is not advisable to use the option `password`, which stores the
+plain text password on the configuration file. But if you do, make sure to store
+it on the local configuration file, which is ignored by Git, so that it does not
+end up being published on GitHub or some other public place.
+
+You can do this by using the option `--local` of `dvc remote modify`:
+
+```dvc
+$ dvc remote modify --local ssh-remote password 12345678
+
+$ cat .dvc/config.local
+['remote "ssh-remote"']
+password = 12345678
+```
+
+</details>
+
+<details>
+
+### Tip: Recommended SSH configuration
```

Comment:
just put an external link, not need to provide the full instruction on how to use SSH keys. Also setup looks too complicated, you would usually just add your pub key to the `authorized_keys` file on the server.

Reply:
No external site would explain it simpler than this. And it is not just sending the pub key to the server, the configuration of ssh (on `~/.ssh/config`) also matters for simplicity. It is thanks to this config file that we don't need to tell DVC neither the username, nor the key file or the port.
So this section is tuned to DVC, and no external site wold explain SSH configuration like this.

Reply:
>  It is thanks to this config file that we don't need to tell DVC neither the username, nor the key file or the port.

that's the point, it's again very advanced very complicated workflow you are describing. You don't use it usually. In 99% you just add your public key to the server. And ssh client is using it by default (I hope that paramiko that we use internally is set to do the regular stuff).

> So this section is tuned to DVC, and no external site wold explain SSH configuration like this.

by default not that much special about DVC and it's enough to name a few options to override key, username, etc.

Reply:
> In 99% you just add your public key to the server.

Using the same key for all the servers is not considered a good practice.
I use different keys for different servers, and even for different services in the same server.

> ssh client is using it by default

ssh client is using it automatically if you tell it in the configuration file (`.ssh/config`) which key to use.

Reply:
@dashohoxha this is the link https://www.digitalocean.com/community/tutorials/how-to-set-up-ssh-keys--2 that should be more than enough. To be honest I'm not even sure if we need even the link. 

> Using the same key for all the servers is not considered a good practice.

one encrypted key is very regular and reasonably secure. Please, share some links that prove your point, similar to one above.

> ssh client is using it automatically if you tell it in the configuration file (.ssh/config) which key to use.

I'm not sure about specific mechanics but I don't remember the last time I needed to make any changes to the config. Again, would love some links with the recommendations on what I am doing wrong,

Reply:
What is the usefulness of recommending a link that everybody can google and find? Besides, it is non-authoritative and already too old.

> one encrypted key is very regular and reasonably secure

This is like using the same password for all the sites.

> Please, share some links that prove your point, similar to one above.

The link that you provided does not prove anything. The links that I may provide do not prove anything. But some reasons why someone might want to use more than a key are on these pages:

- https://www.redhat.com/sysadmin/manage-multiple-ssh-key-pairs
- https://guide.freecodecamp.org/ssh/managing-multiple-ssh-keys/
- https://security.stackexchange.com/questions/40050/best-practice-separate-ssh-key-per-host-and-user-vs-one-ssh-key-for-all-hos

Reply:
> What is the usefulness of recommending a link that everybody can google and find?

Good question. That's exactly the point. No need to include all this information at all. Just mention that we don't recommend using plain text passwords.

> This is like using the same password for all the sites.

Comparing passwords and keys is wrong. They have very different security characteristics.

```
As for universally-applicable guidelines on how to run your security: there are none. The more additional security you add, the more convenience you give up. The one piece of advice I can give categorically is this: keep your private key encrypted. The added security there is pretty significant.
```

Good points.